supervision/detection/track.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+
+from supervision.detection.core import Detections
+from supervision.geometry.core import Position
+
+
+class TrackStorage:
+    """
+    Trace the object trajectory with Detections with tracker id
+    """
+
+    def __init__(
+        self,
+        position: Position = Position.CENTER,
+        max_length: int = 10,
+    ):
+        """
+        Initialize the track storage to store tracked detections from tracker
+        Args:
+            position [sv.Position]: Trace position whethere mid-point or bottom center point
+            max_length [int]: Length of previous detections to annotate
+            Detections objects are stored as numpy ndarray with #counter, x, y, class, track_id
+
+        Example:
+            ```python
+            >>> import supervision as sv
+            >>> track_storage = sv.TrackStorage()
+            >>> for frame in sv.get_video_frames_generator(source_path='source_video.mp4'):
+            >>>     detections = sv.Detections(...)
+            >>>     tracked_objects = tracker(...)
+            >>>     tracked_detections = sv.Detections(tracked_objects)
+            >>>     track_storage.update(tracked_detections)
+        """
+        self.position = position
+        self.max_length = max_length
+        self.frame_counter = 0
+        self.storage = np.zeros((0, 5))
+
+    def update(self, frame_counter: int, detections: Detections) -> None:
+        if detections.xyxy.shape[0] > 0:
+            xyxy = detections.xyxy
+
+            if self.position == Position.CENTER:
```

Comment:
That part of the logic is already implemented in `sv.Detections.get_anchor_coordinates`. Let's refactor our logic here to use the `get_anchor_coordinates` method. Also:

- Let's expand `Position` with:
    -  `TOP_CENTER = "TOP_CENTER"`
    -  `TOP_LEFT = "TOP_LEFT"`
    -  `TOP_RIGHT = "TOP_RIGHT"`
    -  `CENTER_LEFT = "CENTER_LEFT"`
    -  `CENTER_RIGHT = "CENTER_RIGHT"`
    -  `BOTTOM_LEFT = "BOTTOM_LEFT"`
    -  `BOTTOM_RIGHT = "BOTTOM_RIGHT"`

- Let's expand the `get_anchor_coordinates` implementation to handle new `Position`'s
- We need to add unit tests for `get_anchor_coordinates`, but I can handle that. Just let me know when you'll be ready.

Reply:
@SkalskiP , I made changes and added extra anchor points.